### PR TITLE
[ESP32] ESP-IDF v5.0 support in reamining apps

### LINKED ISSUE #24576: [Feature] [ESP32] Support esp-idf v5.0 release
### Feature description

[ESP-IDF Release v5.0](https://github.com/espressif/esp-idf/releases/tag/v5.0) is out some time back and it is not supported in the SDK.

This issue is to track the progress for supporting esp-idf v5.0 alongside esp-idf v4.4.3

### Platform

esp32

### Platform Version(s)

_No response_

### Anything else?
- [x] examples/lighting-app/esp32
- [x] examples/all-clusters-app/esp32
- [x] examples/all-clusters-minimal-app/esp32
- [x] examples/bridge-app/esp32
- [x] examples/light-switch-app/esp32
- [ ] examples/ota-provider-app/esp32
- [ ] examples/ota-requestor-app/esp32
- [ ] examples/temperature-measurement-app/esp32
- [ ] examples/lock-app/esp32
- [ ] examples/chef/esp32

### PR DESCRIPTION
**Pull request by shubhamdp**
####  Problem
Fixes https://github.com/project-chip/connectedhomeip/issues/24576

#### Change Overview
Added support for esp-idf v5.0 in ota-requestor-app, ota-provider-app, temperature-measurement-app, lock-app, and chef-app.

In a nutshell:

- Fixed few format specifier
- Changed ErrorStr to err.Format()
- Added components to required list. In v4.4.4 these made it to list somehow through some other component's global requirement list but, those are not made private and hence needed to be added explicitly.

#### Testing
Successfully compiled above apps for esp32c3 on idf v5.0 and v4.4.3

### CODE REVIEW

File: examples/lock-app/esp32/main/DeviceCallbacks.cpp [hunk 1]
```diff
@@ -59,7 +60,7 @@ void AppDeviceCallbacks::PostAttributeChangeCallback(EndpointId endpointId, Clus
 void AppDeviceCallbacks::OnOnOffPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
 {
     VerifyOrExit(attributeId == app::Clusters::OnOff::Attributes::OnOff::Id,
-                 ESP_LOGI(TAG, "Unhandled Attribute ID: '0x%04x", attributeId));
+                 ESP_LOGI(TAG, "Unhandled Attribute ID: '0x%04" PRIx32, attributeId));
```
Comment (bzbarsky-apple) [inline examples/lock-app/esp32/main/DeviceCallbacks.cpp:63]: Why 04?  I know it was there, but it makes no sense.

File: examples/lock-app/esp32/main/DeviceCallbacks.cpp [hunk 2]
```diff
@@ -39,8 +39,9 @@ using namespace ::chip::DeviceLayer;
 void AppDeviceCallbacks::PostAttributeChangeCallback(EndpointId endpointId, ClusterId clusterId, AttributeId attributeId,
                                                      uint8_t type, uint16_t size, uint8_t * value)
 {
-    ESP_LOGI(TAG, "PostAttributeChangeCallback - Cluster ID: '0x%04x', EndPoint ID: '0x%02x', Attribute ID: '0x%04x'", clusterId,
-             endpointId, attributeId);
+    ESP_LOGI(TAG,
+             "PostAttributeChangeCallback - Cluster ID: '0x%04" PRIx32 "', EndPoint ID: '0x%02x', Attribute ID: '0x%04" PRIx32 "'",
```
Comment (bzbarsky-apple) [inline examples/lock-app/esp32/main/DeviceCallbacks.cpp:43]: Pre-existing, but this code seems to think that cluster/attribute ids are 16-bit and endpoint ids are 8-bit....

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: examples/lock-app/esp32/main/DeviceCallbacks.cpp (hunk 2 above)
Comment (shubhamdp) [reply examples/lock-app/esp32/main/DeviceCallbacks.cpp]: Yeah, fixed these

File: examples/lock-app/esp32/main/DeviceCallbacks.cpp (hunk 1 above)
Comment (shubhamdp) [reply examples/lock-app/esp32/main/DeviceCallbacks.cpp]: Yes, this doesn't make sense, fixed this as well.

**Review by dhrishi — APPROVED** [review]

**Pull request merged**